# Question 1
+ Install Pipenv
+ What's the version of pipenv you installed?
+ Use --version to find out
# Answer: 2023.10.3


In [1]:
!pipenv --version

pipenv, version 2023.10.3


# Question 2
+ Use Pipenv to install Scikit-Learn version 1.3.1
+ What's the first hash for scikit-learn you get in Pipfile.lock?
# Answer: <h1 style="color:red;">e0d12a0331eb3926e2e7c7c37be9118ab3619b518475b3c3897cdb40da98d9d9</h1>

In [8]:
!cat Pipfile.lock

{
    "_meta": {
        "hash": {
            "sha256": "e0d12a0331eb3926e2e7c7c37be9118ab3619b518475b3c3897cdb40da98d9d9"
        },
        "pipfile-spec": 6,
        "requires": {
            "python_version": "3.11"
        },
        "sources": [
            {
                "name": "pypi",
                "url": "https://pypi.org/simple",
                "verify_ssl": true
            }
        ]
    },
    "default": {
        "joblib": {
            "hashes": [
                "sha256:92f865e621e17784e7955080b6d042489e3b8e294949cc44c6eac304f59772b1",
                "sha256:ef4331c65f239985f3f2220ecc87db222f08fd22097a3dd5698f693875f8cbb9"
            ],
            "markers": "python_version >= '3.7'",
            "version": "==1.3.2"
        },
        "numpy": {
            "hashes": [
                "sha256:06934e1a22c54636a059215d6da99e23286424f316fddd979f5071093b648668",
                "sha256:1c59c046c31a43310ad0199d6299e59f57a289e22f0f36951ced1c9eac3665b9",
         

In [10]:
PREFIX='https://raw.githubusercontent.com/DataTalksClub/machine-learning-zoomcamp/master/cohorts/2023/05-deployment/homework'
!wget $PREFIX/model1.bin
!wget $PREFIX/dv.bin

--2023-10-19 08:53:02--  https://raw.githubusercontent.com/DataTalksClub/machine-learning-zoomcamp/master/cohorts/2023/05-deployment/homework/model1.bin
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 842 [application/octet-stream]
Saving to: ‘model1.bin’

model1.bin          100%[===================>]     842  --.-KB/s    in 0s      

2023-10-19 08:53:03 (30.3 MB/s) - ‘model1.bin’ saved [842/842]

--2023-10-19 08:53:03--  https://raw.githubusercontent.com/DataTalksClub/machine-learning-zoomcamp/master/cohorts/2023/05-deployment/homework/dv.bin
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... con

In [12]:
import pickle

In [13]:
dv_file = 'dv.bin'
model_file = 'model1.bin'
with open(dv_file,'rb') as f_in:
    dv = pickle.load(f_in)
with open(model_file,'rb') as f_in:
    model = pickle.load(f_in)

/home/thinker/anaconda3/lib/python3.11/site-packages/sklearn/base.py:318: UserWarning: Trying to unpickle estimator DictVectorizer from version 1.3.1 when using version 1.2.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:
https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
  warnings.warn(
/home/thinker/anaconda3/lib/python3.11/site-packages/sklearn/base.py:318: UserWarning: Trying to unpickle estimator LogisticRegression from version 1.3.1 when using version 1.2.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:
https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
  warnings.warn(


In [15]:
client = {"job": "retired", "duration": 445, "poutcome": "success"}
X = dv.transform([client])
round(model.predict_proba(X)[0,1],3)

0.902

# Question  3: What's the probability that this client will get a credit?
+ 0.162
+ 0.392
+ 0.652
+ 0.902
# A: 0.902


In [16]:
import requests

In [20]:
url = "http://localhost:9690/predict"
client = {"job": "unknown", "duration": 270, "poutcome": "failure"}
response = requests.post(url, json=client).json()

In [21]:
round(response['probability'],3)

0.14

# Question 4: Probability of getting a credit (Flask)
# A: 0.140

# Question 5: So what's the size of this base image?
# A: 147 MB

In [ ]:
url = "YOUR_URL"
client = {"job": "retired", "duration": 445, "poutcome": "success"}
requests.post(url, json=client).json()

# Question 6: What's the probability that this client will get a credit now?
+ 0.168
+ 0.530
+ 0.730
+ 0.968
# A: 